### PR TITLE
Add mechanism to detect Windows version to MSI package

### LINKED ISSUE #3912: Windows agent MSI installer: Detect Windows version
|Wazuh version|Component|Install type|Install method|Platform|
|---|---|---|---|---|
| 3.10 | MSI Installer - SCA | Agent | Packages | Windows |

Currently the only policy installed by default in Windows system is the generic one, this should be changed so that the policy files for the specific running OS are installed as well.

### PR DESCRIPTION
**Pull request by DFolchA**
|Related issue|
|---|
|Closes #3912|

## Description
This PR modifies the WXS file so the MSI package can detect the Windows version and install the SCA files acordingly.

## Tests

- Compilation without warnings in every supported platform
  - [x] Windows
- [x] Package installation
- [x] Package upgrade

### CODE REVIEW

**Review by alberpilot — APPROVED** [review]

**Review by chemamartinez — APPROVED** [review]
LGTM

File: etc/templates/config/windows/2003/syscheck.template [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+    <windows_registry arch="both">HKEY_LOCAL_MACHINE\Software\Microsoft\Active Setup\Installed Components</windows_registry>
+
+    <!-- Windows registry entries to ignore. -->
+    <registry_ignore>HKEY_LOCAL_MACHINE\Security\Policy\Secrets</registry_ignore>
+    <registry_ignore>HKEY_LOCAL_MACHINE\Security\SAM\Domains\Account\Users</registry_ignore>
+    <registry_ignore type="sregex">\Enum$</registry_ignore>
+
+    <!-- Frequency for ACL checking (seconds) -->
+    <windows_audit_interval>300</windows_audit_interval>
```
Comment (chemamartinez) [inline etc/templates/config/windows/2003/syscheck.template:96]: According to the deleted `ossec-pre6.conf` we should review the following changes:

*The audit interval was decreased from 300 to 60 at https://github.com/wazuh/wazuh/pull/4712*
```xml
    <!-- Frequency for ACL checking (seconds) -->
    <windows_audit_interval>60</windows_audit_interval>
```

*The `process_priority`, `max_eps` and `synchronization` block are missing*
```xml 
    <!-- Nice value for Syscheck module -->
    <process_priority>10</process_priority>

    <!-- Maximum output throughput -->
    <max_eps>100</max_eps>

    <!-- Database synchronization settings -->
    <synchronization>
      <enabled>yes</enabled>
      <interval>5m</interval>
      <max_interval>1h</max_interval>
      <max_eps>10</max_eps>
    </synchronization>
```

The same for the Syscheck template for Windows XP

Comment (DFolchA) [reply etc/templates/config/windows/2003/syscheck.template]: Done in a55521c

**Review by chemamartinez — APPROVED** [review]
LGTM

**Review by alberpilot — APPROVED** [review]

**Pull request merged**